bridge/src/main/java/com/livefront/bridge/BridgeDelegate.java

```diff
@@ -208,11 +229,12 @@ public void onActivityCreated(Activity activity, Bundle savedInstanceState) {
 
                         // Make sure we clear all data after creating the first Activity if it does
                         // does not have a saved stated Bundle. (During state restoration, the
-                        // first Activity will always have a non-null saved state Bundle.)
-                        if (!mIsFirstCreateCall) {
+                        // first Activity will always have a non-null saved state Bundle, edge case for
+                        // deeplink call up, it will be non-null for new deeplink activity.)
+                        if (!isFreshStartOrBackToMain(activity, savedInstanceState)) {
                             return;
                         }
-                        mIsFirstCreateCall = false;
```

Comment:
Can you explain why we don't need this check? Shouldn't we still track this and only do the clearing when the app first launches?

Reply:
Hmmm, I merged this flag into check function and then we can change its name to freshStart